### PR TITLE
ci: add reviewer based on label

### PR DESCRIPTION
**Pull request by dundargoc**
Also remove CODEOWNERS since this is a superior solution.

### CODE REVIEW

File: .github/workflows/labeler.yml [hunk 1]
```diff
@@ -25,6 +28,61 @@ jobs:
...
+        run: gh pr edit $PR_NUMBER --add-reviewer smolck
+
+      - if: contains(steps.labels.outputs.labels, 'lsp%')
+        run: gh pr edit $PR_NUMBER --add-reviewer mjlbach,mfussenegger
+
+      - if: contains(steps.labels.outputs.labels, 'lua%')
+        run: gh pr edit $PR_NUMBER --add-reviewer gpanders
+
+      - if: contains(steps.labels.outputs.labels, 'runtime%')
+        run: gh pr edit $PR_NUMBER --add-reviewer seandewar
```
Comment (zeertzjq) [inline .github/workflows/labeler.yml:79]: Are you sure about this? I think the `runtime` label is more likely applied to a PR like #16930.

Comment (clason) [reply .github/workflows/labeler.yml:79]: runtime == funtime :)

(No action by assigned people is _required_; this is is more intended as a service for team members to allow them to stop subscribing to _every_ issue and still get notified of things potentially of interest to them.)

Comment (dundargoc) [reply .github/workflows/labeler.yml:79]: OK. I'll remove it.

Comment (dundargoc) [reply .github/workflows/labeler.yml:79]: @zeertzjq also, are there any labels you want to automatically be a reviewer for?

Comment (zeertzjq) [reply .github/workflows/labeler.yml]: No.

Comment (dundargoc) [reply .github/workflows/labeler.yml]: You sound a bit hesitant. Just let me know if you change your mind.

**Comment by mfussenegger** [comment]
Does this also already add the reviewers on drafts? Would be cool if we can defer the reviewer assignment until a PR is ready for review

**Comment by dundargoc** [comment]
> Does this also already add the reviewers on drafts? Would be cool if we can defer the reviewer assignment until a PR is ready for review

Yes, it currently adds reviewers on draft as well. Adding reviewers after a PR is marked is a good idea, both because the reviewer gets notified when it's actually ready and because we'd have time to adjust any incorrect labels.

**Comment by dundargoc** [comment]
Hmm, this could get a bit messy. The most straightforward way would be to add all logic to the labeler workflow and call it a day, but I'm not entirely convinced this is the way to go. I think we should look into chain workflows together. This would help us separate the workflows better based on their function, and less on convenience. It'll also help us use the same logic for when github-actions is the trigger vs when a person is the trigger.

The annoying part workflow chaining is the lack of documentation and that the syntax somehow seems worse than usual, which is a feat. I'll need to do some research and tinkering with this before I have something of substance.

**Comment by dundargoc** [comment]
@mfussenegger I've thought about it and it's a good idea. There's no reason to stall on it though as this is still a superior solution to what we have today. I can implement your suggestion in a follow-up PR.

File: .github/workflows/labeler.yml [hunk 2]
```diff
@@ -28,3 +30,68 @@ jobs:
...
+        run: gh pr edit $PR_NUMBER --add-reviewer gpanders
+
+      - if: contains(steps.labels.outputs.labels, 'distribution%')
+        run: gh pr edit $PR_NUMBER --add-reviewer jamessan
+
+      - if: contains(steps.labels.outputs.labels, 'documentation%')
+        run: gh pr edit $PR_NUMBER --add-reviewer clason
+
+      - if: contains(steps.labels.outputs.labels, 'extmarks%')
+        run: gh pr edit $PR_NUMBER --add-reviewer muniter
```
Comment (muniter) [inline .github/workflows/labeler.yml:70]: Not really my territory. I only know named marks well :) I guess I can help in the API. I believe most of the work on extmarks has been done by bfredl.

**Review by jamessan — APPROVED** [review]

**Pull request merged**